acvm/src/compiler.rs

```diff
@@ -106,3 +106,61 @@ pub fn compile(
         return_values: acir.return_values,
     })
 }
+
+#[test]
+fn compile_api_compiles_with_backend() {
```

Comment:
I think we can remove this test case as it's just to show that we can pass in a closure here.